### ISSUE TITLE
RIP: 0010:prt_vprintf (lib/vsprintf.c:2626) [e9842826827a2e2a857c3454e23405c5bdd78a38]

### ISSUE THREAD

**Issue by marcin-github** [koverstreet/bcachefs#480]
Hi,
when I copy files from bcachefs to another bcachefs using rsync with "-X" parameter (copying extended attributes) I'm getting:
```
[28840.181703] perf: interrupt took too long (3169 > 3157), lowering kernel.perf_event_max_sample_rate to 63100 
[29337.564595] BUG: kernel NULL pointer dereference, address: 0000000000000008 
[29337.564601] #PF: supervisor read access in kernel mode 
[29337.564604] #PF: error_code(0x0000) - not-present page 
[29337.564606] PGD 0 P4D 0 
[29337.564609] Oops: 0000 [#1] PREEMPT SMP DEBUG_PAGEALLOC 
[29337.564612] CPU: 3 PID: 56169 Comm: rsync Tainted: G     U          T  6.0.0-01801-ge9842826827a #96 
[29337.564617] Hardware name: MSI MS-7982/B150M PRO-VDH (MS-7982), BIOS 3.H0 07/10/2018 
[29337.564619] RIP: 0010:prt_vprintf (lib/vsprintf.c:2626)  
[29337.564623] Code: 01 48 21 ca 0f b6 c0 48 c1 e0 20 48 09 c2 e9 29 ff ff ff 66 0f 1f 84 00 00 00 00 00 41 57 41 56 41 55 41 54 55 53 48 83 ec 50 <44> 8b 47 08 65 48 8b 04 25 28 00 00 00 48 89 44 24 48 31 c0 48 c7 
All code 
======== 
   0:	01 48 21             	add    %ecx,0x21(%rax) 
   3:	ca 0f b6             	lret   $0xb60f 
   6:	c0 48 c1 e0          	rorb   $0xe0,-0x3f(%rax) 
   a:	20 48 09             	and    %cl,0x9(%rax) 
   d:	c2 e9 29             	ret    $0x29e9 
  10:	ff                   	(bad)   
  11:	ff                   	(bad)   
  12:	ff 66 0f             	jmp    *0xf(%rsi) 
  15:	1f                   	(bad)   
  16:	84 00                	test   %al,(%rax) 
  18:	00 00                	add    %al,(%rax) 
  1a:	00 00                	add    %al,(%rax) 
  1c:	41 57                	push   %r15 
  1e:	41 56                	push   %r14 
  20:	41 55                	push   %r13 
  22:	41 54                	push   %r12 
  24:	55                   	push   %rbp 
  25:	53                   	push   %rbx 
  26:	48 83 ec 50          	sub    $0x50,%rsp 
  2a:*	44 8b 47 08          	mov    0x8(%rdi),%r8d		<-- trapping instruction 
  2e:	65 48 8b 04 25 28 00 	mov    %gs:0x28,%rax 
  35:	00 00  
  37:	48 89 44 24 48       	mov    %rax,0x48(%rsp) 
  3c:	31 c0                	xor    %eax,%eax 
  3e:	48                   	rex.W 
  3f:	c7                   	.byte 0xc7 
 
Code starting with the faulting instruction 
=========================================== 
   0:	44 8b 47 08          	mov    0x8(%rdi),%r8d 
   4:	65 48 8b 04 25 28 00 	mov    %gs:0x28,%rax 
   b:	00 00  
   d:	48 89 44 24 48       	mov    %rax,0x48(%rsp) 
  12:	31 c0                	xor    %eax,%eax 
  14:	48                   	rex.W 
  15:	c7                   	.byte 0xc7 
[29337.564629] RSP: 0018:ffffc093c654baf0 EFLAGS: 00010286 
[29337.564632] RAX: ffffc093c654bb98 RBX: ffffc093c654bc18 RCX: 0000000000000000 
[29337.564634] RDX: ffffc093c654bb78 RSI: ffffffff992fea13 RDI: 0000000000000000 
[29337.564636] RBP: ffffc093c654bbc8 R08: 0000000000000000 R09: 0000000000000000 
[29337.564638] R10: 0000000000000000 R11: 0000000000000000 R12: 0000000000000000 
[29337.564640] R13: ffff9ff4f7b01d18 R14: ffff9ff73a000000 R15: 0000000000000009 
[29337.564643] FS:  00007f204bbfc740(0000) GS:ffff9ff81bf80000(0000) knlGS:0000000000000000 
[29337.564646] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033 
[29337.564648] CR2: 0000000000000008 CR3: 00000003bb216006 CR4: 00000000003706e0 
[29337.564650] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000 
[29337.564652] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400 
[29337.564654] Call Trace: 
[29337.564657]  <TASK> 
[29337.564660] prt_printf (lib/vsprintf.c:2781)  
[29337.564663] ? _kstrtoull (lib/kstrtox.c:103)  
[29337.564667] bch2_opt_parse (fs/bcachefs/opts.c:276)  
[29337.564672] bch2_xattr_bcachefs_set (fs/bcachefs/xattr.c:554)  
[29337.564676] ? privileged_wrt_inode_uidgid (kernel/capability.c:494)  
[29337.564680] __vfs_setxattr (fs/xattr.c:182)  
[29337.564685] __vfs_setxattr_noperm (fs/xattr.c:218)  
[29337.564688] vfs_setxattr (./include/linux/fs.h:762 fs/xattr.c:315)  
[29337.564691] setxattr (fs/xattr.c:625)  
[29337.564695] ? __check_object_size (mm/usercopy.c:195 mm/usercopy.c:250 mm/usercopy.c:212)  
[29337.564699] ? strncpy_from_user (./include/asm-generic/access_ok.h:40 ./arch/x86/include/asm/uaccess.h:542 lib/strncpy_from_user.c:138)  
[29337.564702] ? preempt_count_add (./include/linux/ftrace.h:950 kernel/sched/core.c:5624 kernel/sched/core.c:5621 kernel/sched/core.c:5649)  
[29337.564705] ? __mnt_want_write (./arch/x86/include/asm/preempt.h:103 fs/namespace.c:375)  
[29337.564708] path_setxattr (fs/xattr.c:644)  
[29337.564711] __x64_sys_lsetxattr (fs/xattr.c:665 fs/xattr.c:661 fs/xattr.c:661)  
[29337.564714] do_syscall_64 (arch/x86/entry/common.c:50 arch/x86/entry/common.c:80)  
[29337.564719] entry_SYSCALL_64_after_hwframe (arch/x86/entry/entry_64.S:120)  
[29337.564723] RIP: 0033:0x7f204bcfa05a 
[29337.564725] Code: 48 8b 0d c9 9d 0c 00 f7 d8 64 89 01 48 83 c8 ff c3 66 2e 0f 1f 84 00 00 00 00 00 0f 1f 44 00 00 49 89 ca b8 bd 00 00 00 0f 05 <48> 3d 01 f0 ff ff 73 01 c3 48 8b 0d 96 9d 0c 00 f7 d8 64 89 01 48 
All code 
======== 
   0:	48 8b 0d c9 9d 0c 00 	mov    0xc9dc9(%rip),%rcx        # 0xc9dd0 
   7:	f7 d8                	neg    %eax 
   9:	64 89 01             	mov    %eax,%fs:(%rcx) 
   c:	48 83 c8 ff          	or     $0xffffffffffffffff,%rax 
  10:	c3                   	ret     
  11:	66 2e 0f 1f 84 00 00 	cs nopw 0x0(%rax,%rax,1) 
  18:	00 00 00  
  1b:	0f 1f 44 00 00       	nopl   0x0(%rax,%rax,1) 
  20:	49 89 ca             	mov    %rcx,%r10 
  23:	b8 bd 00 00 00       	mov    $0xbd,%eax 
  28:	0f 05                	syscall  
  2a:*	48 3d 01 f0 ff ff    	cmp    $0xfffffffffffff001,%rax		<-- trapping instruction 
  30:	73 01                	jae    0x33 
  32:	c3                   	ret     
  33:	48 8b 0d 96 9d 0c 00 	mov    0xc9d96(%rip),%rcx        # 0xc9dd0 
  3a:	f7 d8                	neg    %eax 
  3c:	64 89 01             	mov    %eax,%fs:(%rcx) 
  3f:	48                   	rex.W 
 
Code starting with the faulting instruction 
=========================================== 
   0:	48 3d 01 f0 ff ff    	cmp    $0xfffffffffffff001,%rax 
   6:	73 01                	jae    0x9 
   8:	c3                   	ret     
   9:	48 8b 0d 96 9d 0c 00 	mov    0xc9d96(%rip),%rcx        # 0xc9da6 
  10:	f7 d8                	neg    %eax 
  12:	64 89 01             	mov    %eax,%fs:(%rcx) 
  15:	48                   	rex.W 
[29337.564730] RSP: 002b:00007ffde58e8988 EFLAGS: 00000206 ORIG_RAX: 00000000000000bd 
[29337.564734] RAX: ffffffffffffffda RBX: 0000000000000000 RCX: 00007f204bcfa05a 
[29337.564736] RDX: 00005613cf21d290 RSI: 00005613cf21d292 RDI: 00007ffde58e8c80 
[29337.564738] RBP: 00005613cf21d292 R08: 0000000000000000 R09: 0000000000000002 
[29337.564740] R10: 0000000000000002 R11: 0000000000000206 R12: 00007f204b4d8010 
[29337.564742] R13: 00005613cf21d670 R14: 0000000000000000 R15: 0000000000000001 
[29337.564745]  </TASK> 
[29337.564747] Modules linked in: wireguard curve25519_x86_64 libcurve25519_generic libchacha20poly1305 chacha_x86_64 poly1305_x86_64 ip6_udp_tunnel udp_tunnel af_packet zram xt_nat xt_tcpudp veth xt_conntrack xt_MASQUERADE nf_conntrack_netlink nfnetlink xfrm_user xfrm_algo xt_addrtype iptable_filter iptable_nat nf_nat nf_conntrack nf_defrag_ipv6 nf_defrag_ipv4 ip_tables x_tables bpfilter br_netfilter nfsd auth_rpcgss nfs_acl nfs lockd grace fscache sunrpc overlay bridge stp llc snd_hda_codec_hdmi i915 binfmt_misc i2c_algo_bit iosf_mbi drm_buddy cec intel_gtt drm_display_helper coretemp hwmon drm_kms_helper cfbfillrect syscopyarea cfbimgblt sysfillrect sysimgblt x86_pkg_temp_thermal intel_powerclamp fb_sys_fops cfbcopyarea fb kvm_intel font pps_ldisc pps_core fbdev snd_hda_intel ttm kvm snd_intel_dspcfg configfs drm snd_hda_codec snd_hda_core snd_pcm snd_timer snd irqbypass rapl i2c_i801 intel_cstate intel_lpss_pci i2c_smbus r8169 realtek mdio_devres of_mdio mei_me fixed_phy fwnode_mdio 
[29337.564808]  mei intel_uncore libphy intel_lpss soundcore i2c_core idma64 cdc_acm drm_panel_orientation_quirks virt_dma bfq tiny_power_button intel_pch_thermal mfd_core video backlight acpi_pad fan button thermal raid1 md_mod xhci_pci crct10dif_pclmul ghash_clmulni_intel xhci_hcd mpt3sas aesni_intel crypto_simd cryptd raid_class scsi_transport_sas dm_mod 
[29337.564840] Unloaded tainted modules: acpi_cpufreq():1 acpi_cpufreq():1 acpi_cpufreq():1 acpi_cpufreq():1 acpi_cpufreq():1 acpi_cpufreq():1 acpi_cpufreq():1 acpi_cpufreq():1 netconsole():1 acpi_cpufreq():1 acpi_cpufreq():1 acpi_cpufreq():1 acpi_cpufreq():1 acpi_cpufreq():1 
[29337.564859] CR2: 0000000000000008 
[29337.564861] ---[ end trace 0000000000000000 ]--- 
[29337.564864] RIP: 0010:prt_vprintf (lib/vsprintf.c:2626)  
[29337.564866] Code: 01 48 21 ca 0f b6 c0 48 c1 e0 20 48 09 c2 e9 29 ff ff ff 66 0f 1f 84 00 00 00 00 00 41 57 41 56 41 55 41 54 55 53 48 83 ec 50 <44> 8b 47 08 65 48 8b 04 25 28 00 00 00 48 89 44 24 48 31 c0 48 c7 
All code 
======== 
   0:	01 48 21             	add    %ecx,0x21(%rax) 
   3:	ca 0f b6             	lret   $0xb60f 
   6:	c0 48 c1 e0          	rorb   $0xe0,-0x3f(%rax) 
   a:	20 48 09             	and    %cl,0x9(%rax) 
   d:	c2 e9 29             	ret    $0x29e9 
  10:	ff                   	(bad)   
  11:	ff                   	(bad)   
  12:	ff 66 0f             	jmp    *0xf(%rsi) 
  15:	1f                   	(bad)   
  16:	84 00                	test   %al,(%rax) 
  18:	00 00                	add    %al,(%rax) 
  1a:	00 00                	add    %al,(%rax) 
  1c:	41 57                	push   %r15 
  1e:	41 56                	push   %r14 
  20:	41 55                	push   %r13 
  22:	41 54                	push   %r12 
  24:	55                   	push   %rbp 
  25:	53                   	push   %rbx 
  26:	48 83 ec 50          	sub    $0x50,%rsp 
  2a:*	44 8b 47 08          	mov    0x8(%rdi),%r8d		<-- trapping instruction 
  2e:	65 48 8b 04 25 28 00 	mov    %gs:0x28,%rax 
  35:	00 00  
  37:	48 89 44 24 48       	mov    %rax,0x48(%rsp) 
  3c:	31 c0                	xor    %eax,%eax 
  3e:	48                   	rex.W 
  3f:	c7                   	.byte 0xc7 
 
Code starting with the faulting instruction 
=========================================== 
   0:	44 8b 47 08          	mov    0x8(%rdi),%r8d 
   4:	65 48 8b 04 25 28 00 	mov    %gs:0x28,%rax 
   b:	00 00  
   d:	48 89 44 24 48       	mov    %rax,0x48(%rsp) 
  12:	31 c0                	xor    %eax,%eax 
  14:	48                   	rex.W 
  15:	c7                   	.byte 0xc7 
[29337.564871] RSP: 0018:ffffc093c654baf0 EFLAGS: 00010286 
[29337.564874] RAX: ffffc093c654bb98 RBX: ffffc093c654bc18 RCX: 0000000000000000 
[29337.564876] RDX: ffffc093c654bb78 RSI: ffffffff992fea13 RDI: 0000000000000000 
[29337.564878] RBP: ffffc093c654bbc8 R08: 0000000000000000 R09: 0000000000000000 
[29337.564880] R10: 0000000000000000 R11: 0000000000000000 R12: 0000000000000000 
[29337.564882] R13: ffff9ff4f7b01d18 R14: ffff9ff73a000000 R15: 0000000000000009 
[29337.564884] FS:  00007f204bbfc740(0000) GS:ffff9ff81bf80000(0000) knlGS:0000000000000000 
[29337.564887] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033 
[29337.564889] CR2: 0000000000000008 CR3: 00000003bb216006 CR4: 00000000003706e0 
[29337.564891] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000 
[29337.564893] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400 
```

**Comment by koverstreet** [comment]
Fixed by ea18ecafac

**Comment by marcin-github** [comment]
Thanks!
